src/Microsoft.Health.Encryption/Customer/Health/CustomerKeyValidationBackgroundService.cs

```diff
@@ -0,0 +1,105 @@
+using Azure;
+using EnsureThat;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Core.Features.Health;
+using Microsoft.Health.Encryption.Customer.Configs;
+
+namespace Microsoft.Health.Encryption.Customer.Health;
+
+internal class CustomerKeyValidationBackgroundService : BackgroundService
+{
+    private const string AccessLostMessage = "Access to the customer-managed key has been lost";
+    private const int DefaultKeyValidationPeriodInSeconds = 60;
+
+    private readonly IKeyTestProvider _keyTestProvider;
+    private readonly ValueCache<CustomerKeyHealth> _customerManagedKeyHealth;
+    private readonly ILogger<CustomerKeyValidationBackgroundService> _logger;
+
+    private readonly TimeSpan _keyValidationPeriod;
+
+    public CustomerKeyValidationBackgroundService(
+        IKeyTestProvider keyTestProvider,
+        ValueCache<CustomerKeyHealth> customerManagedKeyHealth,
+        IOptions<CustomerManagedKeyOptions> customerManagedKeyOptions,
+        ILogger<CustomerKeyValidationBackgroundService> logger)
+    {
+        EnsureArg.IsNotNull(customerManagedKeyOptions, nameof(customerManagedKeyOptions));
+
+        _keyTestProvider = EnsureArg.IsNotNull(keyTestProvider, nameof(keyTestProvider));
+        _customerManagedKeyHealth = EnsureArg.IsNotNull(customerManagedKeyHealth, nameof(customerManagedKeyHealth));
+        _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+
+        _keyValidationPeriod = customerManagedKeyOptions.Value != null && customerManagedKeyOptions.Value.KeyValidationPeriodInSeconds > 0 ?
+            TimeSpan.FromSeconds(customerManagedKeyOptions.Value.KeyValidationPeriodInSeconds) :
+            TimeSpan.FromSeconds(DefaultKeyValidationPeriodInSeconds);
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            try
+            {
+                await CheckHealth(stoppingToken).ConfigureAwait(false);
+                await Task.Delay(_keyValidationPeriod, stoppingToken).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException e)
+            {
+                _logger.LogInformation(e, $"{nameof(CustomerKeyValidationBackgroundService)} cancelled");
+            }
+        }
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1031:Do not catch general exception types", Justification = "This background service failing should not crash the application.")]
+    internal async Task CheckHealth(CancellationToken cancellationToken)
+    {
+        try
+        {
+            await _keyTestProvider.AssertHealthAsync(cancellationToken).ConfigureAwait(false);
+            SetHealthy();
+        }
+        catch (Exception ex) when (ex is RequestFailedException || ex is CryptographicException || ex is InvalidOperationException || ex is NotSupportedException)
```

Comment:
nit: use pattern-matching!
```suggestion
        catch (Exception ex) when (ex is RequestFailedException or CryptographicException or InvalidOperationException or NotSupportedException)
```